probatus/missing_values/imputation.py

```diff
@@ -0,0 +1,384 @@
+                remainder="passthrough",
+            )
+
+            model_pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor), ("classifier", self.clf)]
+            )
+
+            temp_results = self._calculate_results(
+                X, y, clf=model_pipeline, strategy="No Imputation"
+            )
+            results.append(temp_results)
+
+        self.report_df = pd.DataFrame(results)
+        self.report_df.sort_values(by=f"{self.scorer.metric_name} score", inplace=True)
+        self.fitted = True
+        return self
+
+    def _calculate_results(self, X, y, clf, strategy):
+        """
+        Method to calculate the results for a particular imputation strategy.
+
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            clf (binary classifier,sklearn.Pipeline):
+                A binary classification model, that will used to evaluate various imputation strategies.
+
+            strategy(string):
+                Name of the strategy used for imputation.
+
+        Returns:
+
+            temp_df(dict) :  Dictionary containing the results of the evaluation.
+        """
+        imputation_results = cross_val_score(
+            clf, X, y, scoring=self.scorer.scorer, cv=self.cv, n_jobs=self.n_jobs
+        )
+
+        temp_results = {
+            "strategy": strategy,
+            f"{self.scorer.metric_name} score": np.round(
+                np.mean(imputation_results), 3
+            ),
+            f"{self.scorer.metric_name} std": np.round(np.std(imputation_results), 3),
+        }
+
+        return temp_results
+
+    def compute(self):
+        """
+        Checks if fit() method has been run and computes the DataFrame with results of imputation for each
+        strategy.
+
+        Returns:
+            (pd.DataFrame):
+                 DataFrame with results of imputation for each strategy.
+        """
+        self._check_if_fitted()
+        return self.report_df
+
+    def fit_compute(self, X, y, column_names=None):
+        """
+        Calculates the cross validated results for various imputation strategies.
+
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            column_names (None, or list of str, optional):
+                List of feature names for the dataset.
+                If None, then column names from the X dataframe are used.
+
+        Returns:
+            (pd.DataFrame):
+                DataFrame with results of imputation for each strategy.
+
+        """
+        self.fit(X, y, column_names=column_names)
+        return self.compute()
+
+    def plot(self, show=True, **figure_kwargs):
+        """
+        Generates plot of the performance of various imputation strategies.
+
+        Args:
+            show (bool, optional):
```

Comment:
If you also compute the train scores, consider adding parameter target_set so the user can select whether to plot train scores, or test/validation scores. This also tells a lot about overfitting.

Like here: https://github.com/ing-bank/probatus/blob/main/probatus/interpret/model_interpret.py

Reply:
true, I updated the plots to show the train results as well. However instead of allowing it as parameter, to keep it simple the cross validated train and cv-test are both shown.